Unable to create native debug build in Gradle

### Describe the bug

I wanted to create a native debug build in Gradle. I attached the following arguments to the 'build' task:
`-Dquarkus.package.type=native -Dquarkus.native.debug.enabled=true`


However, the following exception happens at the end of the build:

```
> io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: java.lang.RuntimeException: Failed to build native image
        at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:223)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:820)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2442)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1476)
        at java.base/java.lang.Thread.run(Thread.java:829)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
  Caused by: java.nio.file.NoSuchFileException: /Users/user/quarkus-backend/backend/build/backend-1.0.0-SNAPSHOT-native-image-source-jar/sources
        at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
        at java.base/sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
        at java.base/sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:149)
        at java.base/java.nio.file.Files.readAttributes(Files.java:1764)
        at java.base/java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:225)
        at java.base/java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:276)
        at java.base/java.nio.file.FileTreeWalker.walk(FileTreeWalker.java:322)
        at java.base/java.nio.file.Files.walkFileTree(Files.java:2717)
        at io.quarkus.bootstrap.util.IoUtils.copy(IoUtils.java:126)
        at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:213)
```

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

Darwin hostname 20.6.0 Darwin Kernel Version 20.6.0: Wed Jun 23 00:26:31 PDT 2021; root:xnu-7195.141.2~5/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk version "11.0.12" 2021-07-20 OpenJDK Runtime Environment GraalVM CE 21.2.0 (build 11.0.12+6-jvmci-21.2-b08) OpenJDK 64-Bit Server VM GraalVM CE 21.2.0 (build 11.0.12+6-jvmci-21.2-b08, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.0.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

------------------------------------------------------------ Gradle 7.1.1 ------------------------------------------------------------  Build time:   2021-07-02 12:16:43 UTC Revision:     774525a055494e0ece39f522ac7ad17498ce032c  Kotlin:       1.4.31 Groovy:       3.0.7 Ant:          Apache Ant(TM) version 1.10.9 compiled on September 27 2020 JVM:          11.0.12 (GraalVM Community 11.0.12+6-jvmci-21.2-b08) OS:           Mac OS X 11.5 x86_64

### Additional information

_No response_

I just tried to download a new project from code.quarkus.io. Here the problem happens too.

```
Execution failed for task ':quarkusBuild'.
> io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: java.lang.RuntimeException: Failed to build native image
        at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:227)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:820)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2442)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1476)
        at java.base/java.lang.Thread.run(Thread.java:829)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
  Caused by: java.nio.file.NoSuchFileException: /Users/user/Downloads/code-with-quarkus/build/code-with-quarkus-1.0.0-SNAPSHOT-native-image-source-jar/sources
        at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
        at java.base/sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
        at java.base/sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:149)
        at java.base/java.nio.file.Files.readAttributes(Files.java:1764)
        at java.base/java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:225)
        at java.base/java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:276)
        at java.base/java.nio.file.FileTreeWalker.walk(FileTreeWalker.java:322)
        at java.base/java.nio.file.Files.walkFileTree(Files.java:2717)
        at io.quarkus.bootstrap.util.IoUtils.copy(IoUtils.java:126)
        at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:217)
        ... 11 more

```

--- a/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRunner.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRunner.java
@@ -45,8 +45,8 @@ public abstract class NativeImageBuildRunner {
     public void addShutdownHook(Process buildNativeProcess) {
     }
 
-    public int build(List<String> args, String nativeImageName, String resultingExecutableName, Path outputDir,
-            boolean debugEnabled, boolean processInheritIODisabled)
+    public Result build(List<String> args, String nativeImageName, String resultingExecutableName, Path outputDir,
+            boolean debugSymbolsEnabled, boolean processInheritIODisabled)
             throws InterruptedException, IOException {
         preBuild(args);
         try {
@@ -63,22 +63,26 @@ public abstract class NativeImageBuildRunner {
             executor.shutdown();
             errorReportLatch.await();
             int exitCode = process.waitFor();
+            boolean objcopyExists = objcopyExists();
             if (exitCode != 0) {
-                return exitCode;
+                return new Result(exitCode, objcopyExists);
             }
 
-            if (objcopyExists()) {
-                if (debugEnabled) {
+            if (objcopyExists) {
+                if (debugSymbolsEnabled) {
                     splitDebugSymbols(nativeImageName, resultingExecutableName);
                 } else {
                     // Strip debug symbols regardless, because the underlying JDK might contain them
                     objcopy("--strip-debug", resultingExecutableName);
                 }
             } else {
-                log.warn("objcopy executable not found in PATH. Debug symbols will not be separated from executable.");
-                log.warn("That will result in a larger native image with debug symbols embedded in it.");
+                if (!debugSymbolsEnabled) {
+                    log.warn(
+                            "objcopy executable not found in PATH. Debug symbols will therefore not be separated from the executable.");
+                    log.warn("That also means that resulting native executable is larger as it embeds the debug symbols.");
+                }
             }
-            return 0;
+            return new Result(0, objcopyExists);
         } finally {
             postBuild();
         }
@@ -141,6 +145,24 @@ public abstract class NativeImageBuildRunner {
             if (process != null) {
                 process.destroy();
             }
+        }
+    }
+
+    static class Result {
+        private final int exitCode;
+        private final boolean objcopyExists;
+
+        public Result(int exitCode, boolean objcopyExists) {
+            this.exitCode = exitCode;
+            this.objcopyExists = objcopyExists;
+        }
+
+        public int getExitCode() {
+            return exitCode;
+        }
+
+        public boolean isObjcopyExists() {
+            return objcopyExists;
         }
     }
 }

--- a/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java
@@ -198,24 +198,32 @@ public class NativeImageBuildStep {
 
             List<String> nativeImageArgs = commandAndExecutable.args;
 
-            int exitCode = buildRunner.build(nativeImageArgs, nativeImageName, resultingExecutableName, outputDir,
+            NativeImageBuildRunner.Result buildNativeResult = buildRunner.build(nativeImageArgs, nativeImageName,
+                    resultingExecutableName, outputDir,
                     nativeConfig.debug.enabled, processInheritIODisabled.isPresent());
-            if (exitCode != 0) {
-                throw imageGenerationFailed(exitCode, nativeImageArgs);
+            if (buildNativeResult.getExitCode() != 0) {
+                throw imageGenerationFailed(buildNativeResult.getExitCode(), nativeImageArgs);
             }
             IoUtils.copy(generatedExecutablePath, finalExecutablePath);
             Files.delete(generatedExecutablePath);
             if (nativeConfig.debug.enabled) {
-                final String symbolsName = String.format("%s.debug", nativeImageName);
-                Path generatedSymbols = outputDir.resolve(symbolsName);
-                Path finalSymbolsPath = outputTargetBuildItem.getOutputDirectory().resolve(symbolsName);
-                IoUtils.copy(generatedSymbols, finalSymbolsPath);
-                Files.delete(generatedSymbols);
-                final String sources = "sources";
-                final Path generatedSources = outputDir.resolve(sources);
-                final Path finalSources = outputTargetBuildItem.getOutputDirectory().resolve(sources);
-                IoUtils.copy(generatedSources, finalSources);
-                IoUtils.recursiveDelete(generatedSources);
+                if (buildNativeResult.isObjcopyExists()) {
+                    final String symbolsName = String.format("%s.debug", nativeImageName);
+                    Path generatedSymbols = outputDir.resolve(symbolsName);
+                    Path finalSymbolsPath = outputTargetBuildItem.getOutputDirectory().resolve(symbolsName);
+                    IoUtils.copy(generatedSymbols, finalSymbolsPath);
+                    Files.delete(generatedSymbols);
+                    final String sources = "sources";
+                    final Path generatedSources = outputDir.resolve(sources);
+                    final Path finalSources = outputTargetBuildItem.getOutputDirectory().resolve(sources);
+                    IoUtils.copy(generatedSources, finalSources);
+                    IoUtils.recursiveDelete(generatedSources);
+                } else {
+                    log.warn(
+                            "objcopy executable not found in PATH. Debug symbols therefore cannot be placed into the dedicated directory.");
+                    log.warn("That also means that resulting native executable is larger as it embeds the debug symbols.");
+                }
+
             }
             System.setProperty("native.image.path", finalExecutablePath.toAbsolutePath().toString());
 

